source/base/DateTime.ooc

```diff
@@ -14,22 +14,232 @@
+		this compareTo(other) == Order equal
+	}
+	operator != (other: This) -> Bool {
+		this compareTo(other) != Order equal
+	}
+	operator < (other: This) -> Bool {
+		this compareTo(other) == Order less
+	}
+	operator <= (other: This) -> Bool {
+		this compareTo(other) != Order greater
+	}
+	operator > (other: This) -> Bool {
+		this compareTo(other) == Order greater
+	}
+	operator >= (other: This) -> Bool {
+		this compareTo(other) != Order less
+	}
+
+	/* number of days in year ( non-leap ) */
+	daysPerYear: static const Int = 365
+	daysPerFourYears: static const Int = 3 * This daysPerYear + 366
+	ticksPerMillisecond: static const Int64 = 1000
+	ticksPerSecond: static const Int64 = This ticksPerMillisecond * 1000
+	ticksPerMinute: static const Int64 = This ticksPerSecond * 60
+	ticksPerHour: static const Int64 = This ticksPerMinute * 60
+	ticksPerDay: static const Int64 = This ticksPerHour * 24
+	ticksPerWeek: static const UInt64 = This ticksPerDay * 7
+	ticksPerFourYears: static const Int64 = This daysPerFourYears * This ticksPerDay
+	/* default date/time printing format */
+	defaultFormat: static const String = "%yyyy-%MM-%dd %hh:%mm:%ss::%zzzz"
+
+	now: static DateTime {
+		get {
+					data := Time dateTimeData()
+					DateTime new(data year, data month, data day, data hour, data minute, data second, data millisecond)
+		}
+	}
+
+	_ticksToDateTimeHelper: static func (totalTicks: Int64) -> DateTimeData {
+		fourYearBlocks := totalTicks / This ticksPerFourYears
+		year := 4 * fourYearBlocks
+		ticksLeft := totalTicks - fourYearBlocks * This ticksPerFourYears
+		for (y in year + 1 .. year + 5) {
+			t := This ticksInYear(y)
+			if (ticksLeft < t) {
+				year = y
+				break
+			} else
+				ticksLeft -= t
+		}
+		month := 0
+		for (m in 1 .. 13) {
+			t := This ticksInMonth(year, m)
+			if (ticksLeft < t) {
+				month = m
+				break
+			} else
+				ticksLeft -= t
+		}
+		days := ticksLeft / This ticksPerDay
+		ticksLeft -= days * This ticksPerDay
+		hour := ticksLeft / This ticksPerHour
+		ticksLeft -= hour * This ticksPerHour
+		minute := ticksLeft / This ticksPerMinute
+		ticksLeft -= minute * This ticksPerMinute
+		second := ticksLeft / This ticksPerSecond
+		ticksLeft -= second * This ticksPerSecond
+		millisecond := ticksLeft / This ticksPerMillisecond
+		DateTimeData new(year, month, days + 1, hour, minute, second, millisecond)
+	}
+
+	/* returns number of ticks for given hours, minutes and seconds */
+	timeToTicks: static func (hours, minutes, seconds, millisecond: Int) -> Int64 {
+		(hours * 3600 + minutes * 60 + seconds) * This ticksPerSecond + millisecond * This ticksPerMillisecond
+	}
+
+	/* returns number of ticks for given date at 0:00*/
+	dateToTicks: static func(year, month, day : Int) -> UInt64 {
+		result := 0 as UInt64
+		if (This dateIsValid(year, month, day)) {
+			totalDays := day - 1
+			for (m in 1 .. month)
+				totalDays += This daysInMonth(year, m)
+			fourYearBlocks := (year - 1) / 4
+			year_start := fourYearBlocks * 4
```

Comment:
I'm not sure what this variable signifies, but it should be `startYear` or `yearStart`.